tests/core/v5_1/alexandria/test_alexandria_client.py

```diff
@@ -48,6 +49,60 @@ async def _respond_empty():
                 )
 
 
+@pytest.mark.trio
+async def test_request_ensures_valid_msg_type(bob, alice_alexandria_client):
+    with pytest.raises(TypeError):
+        await alice_alexandria_client._request(
+            bob.node_id,
+            bob.endpoint,
+            PongMessage(PongPayload(enr_seq=1234, advertisement_radius=4321)),
+            PongMessage,
+            request_id=b"\x01\x02",
+        )
+
+
+@pytest.mark.trio
+async def test_send_response_ensures_valid_msg_type(bob, alice_alexandria_client):
+    with pytest.raises(TypeError):
+        await alice_alexandria_client._send_response(
+            bob.node_id,
+            bob.endpoint,
+            PingMessage(PingPayload(enr_seq=1234, advertisement_radius=4321)),
+            request_id=b"\x01\x02",
+        )
+
+
+@pytest.mark.trio
+async def test_feed_talk_requests_ignores_wrongly_typed_msgs(
```

Comment:
We're missing a similar test for `_feed_talk_responses`. I'll add one if you're happy with the approach I used here to fix the issue